pkg/init/init.go

```diff
@@ -22,13 +22,19 @@ import (
 func copyMetadata(info os.FileInfo, path string) error {
 	// would rather use fd than path but Go makes this very difficult at present
 	stat := info.Sys().(*syscall.Stat_t)
-	if err := syscall.Lchown(path, int(stat.Uid), int(stat.Gid)); err != nil {
+	if err := unix.Lchown(path, int(stat.Uid), int(stat.Gid)); err != nil {
 		return err
 	}
 	timespec := []unix.Timespec{unix.Timespec(stat.Atim), unix.Timespec(stat.Mtim)}
 	if err := unix.UtimesNanoAt(unix.AT_FDCWD, path, timespec, unix.AT_SYMLINK_NOFOLLOW); err != nil {
 		return err
 	}
+	// after chown suid bits may be dropped; also setting them with os.OpenFile is not well defined
+	if info.Mode().IsRegular() {
```

Comment:
amended to be not symlinks (you cant set mode on a symlink without following in Linux)